packages/components/src/Checkbox/Checkbox.test.tsx

```diff
@@ -39,6 +40,37 @@ it("renders each variation of checked, defaultChecked and indeterminate", () =>
   });
 });
 
+it("renders a description when set", () => {
+  const { getByText } = render(<Checkbox description="Checkers" />);
+  const description = getByText("Checkers");
+  expect(description).toBeInstanceOf(HTMLParagraphElement);
+});
+
+it("allows react components to be passed as children", () => {
+  const handleLinkClick = jest.fn();
+  const { getByText } = render(
+    <Checkbox description="Checkers">
+      <Text>
+        I agree to the
+        <a
+          style={{
+            color: "var(--color-interactive)",
+            textDecoration: "underline",
+          }}
+          tabIndex={0}
+          onClick={handleLinkClick}
+        >
+          Terms of Service
+        </a>
+      </Text>
+    </Checkbox>,
+  );
+  const TOSAnchorElement = getByText("Terms of Service");
+  expect(TOSAnchorElement).toBeInTheDocument();
+  fireEvent.click(TOSAnchorElement);
+  expect(handleLinkClick).toHaveBeenCalled();
```

Comment:
interesting. is this to check if the onclick on a child still fires inside a checkbox? if so, when would it not be?

Reply:
Yeah, that was just me being very pedantic with ensuring that the child component behaves how I want it to. I've looked at other components that accept children and their tests regarding children components seem to not go too far. For example, I looked at `Modal.test.tsx` and it seems that what matters to them is that the child gets rendered and that's it.

This kind of relates to the comment that @jackson-zhipeng-chang left. I don't know how far we normally go with our tests as this is my first time modifying an Atlantis component, so I was trying to be very detailed.

Reply:
Ok yeah, just wanted to know the reason behind it. It was odd at first because it looked like you're testing if the onclick fires on the `a` tag but that's already a tried and true thing for raw HTML tags. 

That to me seems like a test that you would do when you do use it in a component that does exactly that interaction. On the other hand tho, it is a good test to see that the onClick within the children still fires if the parent does eventually prevent it from firing. Just making sure that I understand what you're doing and see if there's another way to test it/make it clearer by separating it on it's own `it`.

If you think this is a good test, then yeah, maybe put it on another `it("should still fire the onClick within the children")` or something like that so it's clear what exactly it's for.

Reply:
Good call out, I separated the tests so that there is one for ensuring `onClick` works within the children, and another for rendering a non-empty child